### PR TITLE
add AutoDeleteOnIdle setting

### PR DESCRIPTION
**Pull request by steve-haar**
exposes AutoDeleteOnIdle setting for queues and topics.

closes https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/issues/358

### CODE REVIEW

File: src/Transport/AzureServiceBusSettings.cs [hunk 1]
```diff
@@ -96,6 +96,19 @@ public AzureServiceBusSettings EntityMaximumSize(int maximumSizeInGB)
             return this;
         }
 
+        /// <summary>
+        /// Overrides the default time to auto delete queues and topics when idle.
+        /// </summary>
+        [ObsoleteEx(
```
Comment (SeanFeldman) [inline src/Transport/AzureServiceBusSettings.cs:102]: The setting (if added) shouldn't be obsoleted.

File: src/Transport/AzureServiceBusTransport.cs [hunk 2]
```diff
@@ -110,6 +110,26 @@ public int EntityMaximumSize
...
+        /// <summary>
+        /// The <see cref="TimeSpan"/> idle interval after which topics and queues are automatically deleted.
+        /// </summary>
+        /// <remarks>The minimum duration is 5 minutes. Default value is <see cref="TimeSpan.MaxValue"/>.</remarks>
+        public TimeSpan AutoDeleteOnIdle
+        {
+            get => autoDeleteOnIdle;
+            set
+            {
+                if (value < TimeSpan.FromMinutes(5))
```
Comment (SeanFeldman) [inline src/Transport/AzureServiceBusTransport.cs:122]: Worth adding a comment with a link to the docs about the minimal value of `QueueDescription.AutoDeleteOnIdle`.

File: src/Transport/AzureServiceBusSettings.cs (hunk 1 above)
Comment (steve-haar) [reply src/Transport/AzureServiceBusSettings.cs]: removed setting from `AzureServiceBusSettings`

File: src/Transport/AzureServiceBusTransport.cs (hunk 2 above)
Comment (steve-haar) [reply src/Transport/AzureServiceBusTransport.cs:123]: added comment with link to microsoft docs

**Pull request closed without merging**